telegram/ext/_dictpersistence.py

```diff
@@ -315,8 +312,8 @@ def update_user_data(self, user_id: int, data: Dict) -> None:
             data (:obj:`dict`): The :attr:`telegram.ext.Dispatcher.user_data` ``[user_id]``.
         """
         if self._user_data is None:
-            self._user_data = defaultdict(dict)
-        if self._user_data.get(user_id) == data:
+            self._user_data = {}
+        if self._user_data.get(user_id, None) == data:
```

Comment:
```suggestion
        if self._user_data.get(user_id) == data:
```

Deepsource actually complains about this, since `None` is the default for the second argument :)

Reply:
oh I thought it raised an error like `.pop()`